### ISSUE TITLE
Error when throwing render() or redirect() from client-side onBeforeRender()

### ISSUE THREAD

**Issue by lfos** [vikejs/vike#1910]
### Description

Calling `throw render("/path")` or `throw redirect("/path")`  from my client-side onBeforeRender hook results in an error on first render (but works fine when reaching the same page using client-side navigation):

```
createErrorWithCleanStackTrace.js:4 Uncaught (in promise) Error: [vike@0.4.193][Bug] You stumbled upon a Vike bug. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error. A maintainer will fix the bug (usually under 24 hours).
    at createErrorWithCleanStackTrace (createErrorWithCleanStackTrace.js:4:17)
    at assert (assert.js:54:27)
    at renderPageClientSide (renderPageClientSide.js:27:5)
    at renderErrorPage (renderPageClientSide.js:223:27)
    at async onError (renderPageClientSide.js:37:13)
    at async renderPageNominal (renderPageClientSide.js:142:17)
    at async renderPageClientSide (renderPageClientSide.js:33:5)
    at async installClientRouter (installClientRouter.js:17:5)
```

`onBeforeRender` is configured to run on the client side via `meta`.

For context, this is an SSG project.

**Comment by lfos** [comment]
This doesn't seem to be isolated to onBeforeRender; I am getting the same error when I move the code to `+data.client.ts`.

**Comment by brillout** [comment]
Can you try again with the latest version? If you still get the error then what's the `assert()` call `at renderPageClientSide (renderPageClientSide.js:27:5)` (the third line of the stack trace)?

**Comment by lfos** [comment]
> Can you try again with the latest version? If you still get the error then what's the `assert()` call `at renderPageClientSide (renderPageClientSide.js:27:5)` (the third line of the stack trace)?

This is still happening with version 0.4.198.

The assert() call is `assert(isClientSideNavigation === !isFirstRender)`.  Adding a breakpoint there, it looks like the problem is that when the page first loads, `isClientSideNavigation` is false and `isFirstRender` is true -- but when the assert is reached a second time (likely after the render/redirect),  `isClientSideNavigation` and `isFirstRender` are both false.

**Comment by brillout** [comment]
Fix pre-released as `0.4.198-commit-05a4973`.
